src/Compiler/Interactive/fsi.fs

```diff
@@ -828,10 +828,9 @@ type internal FsiCommandLineOptions(fsi: FsiEvaluationSessionHostConfig,
                 fsiConsoleOutput: FsiConsoleOutput) =
 
     let mutable enableConsoleKeyProcessing =
-       // Mono on Win32 doesn't implement correct console processing
-       not (runningOnMono && Environment.OSVersion.Platform = PlatformID.Win32NT)
+       not (Environment.OSVersion.Platform = PlatformID.Win32NT)
```

Comment:
`runningOnMono` is always false, so this should have become `let mutable enableConsoleKeyProcessing = true`